src/Fixer/Import/GroupImportFixer.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Import;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\NamespaceUseAnalysis;
+use PhpCsFixer\Tokenizer\Analyzer\NamespaceUsesAnalyzer;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Vladimir Kuprienko <vldmr.kuprienko@gmail.com>
+ */
+final class GroupImportFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'There MUST be group use for the same namespaces.',
+            [
+                new VersionSpecificCodeSample(
+                    "<?php\nuse Foo\\Bar;\nuse Foo\\Baz;\n",
+                    new VersionSpecification(70000)
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return \PHP_VERSION_ID >= 70000 && $tokens->isTokenKindFound(T_USE);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $useWithSameNamespaces = $this->getSameNamespaces($tokens);
+
+        if (empty($useWithSameNamespaces)) {
+            return;
+        }
+
+        $this->removeSingleUseStatements($useWithSameNamespaces, $tokens);
+        $this->addGroupUseStatements($useWithSameNamespaces, $tokens);
+    }
+
+    /**
+     * Gets namespace use analyzers with same namespaces.
+     *
+     * @param Tokens $tokens
+     *
+     * @return array|NamespaceUseAnalysis[]
+     */
+    private function getSameNamespaces(Tokens $tokens)
+    {
+        $useDeclarations = (new NamespaceUsesAnalyzer())->getDeclarationsFromTokens($tokens);
+
+        if (0 === \count($useDeclarations)) {
+            return [];
+        }
+
+        $allNamespaces = array_map([$this, 'getNamespaceNameWithSlash'], $useDeclarations);
```

Comment:
please replace the call-by-name with an anonymous function here to call `$this->getNamespaceNameWithSlash`

Reply:
@SpacePossum what's the difference?

Reply:
We like to do use explicit call to functions and methods in this project. Beside having one way of doing things, the main reason is having better SCA support, which helps out with refactoring, clean ups, reporting and other such things based on that analyse. I hope one of the other members of this project can explain it better, I'm not good at that.

Reply:
@SpacePossum ok, I got it. Fixed!